### PR TITLE
Allow some apps to have root URLs in their own routing file

### PR DESCRIPTION
**Pull request by nickvergessen**
Follow up to prevent #20038  from spreading

### CODE REVIEW

File: core/routes.php [hunk 1]
```diff
@@ -85,18 +85,6 @@
...
-		['name' => 'sharecontroller#showShare', 'url' => '/s/{token}', 'verb' => 'GET', 'app' => 'files_sharing'],
-		['name' => 'sharecontroller#showAuthenticate', 'url' => '/s/{token}/authenticate/{redirect}', 'verb' => 'GET', 'app' => 'files_sharing'],
-		['name' => 'sharecontroller#authenticate', 'url' => '/s/{token}/authenticate/{redirect}', 'verb' => 'POST', 'app' => 'files_sharing'],
-		['name' => 'sharecontroller#downloadShare', 'url' => '/s/{token}/download', 'verb' => 'GET', 'app' => 'files_sharing'],
-		['name' => 'publicpreview#directLink', 'url' => '/s/{token}/preview', 'verb' => 'GET', 'app' => 'files_sharing'],
-		['name' => 'requesthandlercontroller#addShare', 'url' => '/ocm/shares', 'verb' => 'POST', 'app' => 'cloud_federation_api'],
-		['name' => 'requesthandlercontroller#receiveNotification', 'url' => '/ocm/notifications', 'verb' => 'POST', 'app' => 'cloud_federation_api'],
-		['name' => 'pagecontroller#showCall', 'url' => '/call/{token}', 'verb' => 'GET', 'app' => 'spreed'],
-		['name' => 'pagecontroller#authenticatePassword', 'url' => '/call/{token}', 'verb' => 'POST', 'app' => 'spreed'],
```
Comment (nickvergessen) [inline core/routes.php]: Those are now in https://github.com/nextcloud/spreed/pull/3134

All other apps are in this PR

File: lib/private/AppFramework/Routing/RouteConfig.php [hunk 2]
```diff
@@ -98,42 +106,7 @@ public function register() {
 	private function processOCS(array $routes): void {
 		$ocsRoutes = $routes['ocs'] ?? [];
 		foreach ($ocsRoutes as $ocsRoute) {
-			$name = $ocsRoute['name'];
```
Comment (nickvergessen) [inline lib/private/AppFramework/Routing/RouteConfig.php]: Best review this file on https://github.com/nextcloud/server/pull/20114/files?w=1

**Review by ChristophWurst — APPROVED** [review]

**Review by rullzer — APPROVED** [review]

**Comment by skjnldsv** [comment]
Rebased

**Comment by skjnldsv** [comment]
Lots of failures

**Review by danxuliu — CHANGES_REQUESTED** [review]
PHP CS is not happy.

Some integration and acceptance tests failures are legit, and all seem to be caused by `Unable to generate a URL for the named route \"cloud_federation_api.RequestHandler.addShare\" as such route does not exist.`

**Comment by nickvergessen** [comment]
> PHP CS is not happy.

Should be fixed by a last rebase as it's not caused by this PR.

**Comment by nickvergessen** [comment]
SHould be good to go now?
Not sure what the acceptance tests complain about. And the failing integration tests are caused by #16035

**Comment by rullzer** [comment]
> SHould be good to go now?
> Not sure what the acceptance tests complain about. And the failing integration tests are caused by #16035

Lets do this and we fix stuff if it does boom

**Comment by danxuliu** [comment]
> Not sure what the acceptance tests complain about

The _app-comments_ can be ignored, they are caused by a regression in the sidebar. The others are legit; for some reason with this pull request the Apps app and the Users app show a blank page (and in [8f650fe^](https://github.com/nextcloud/server/commit/613f0f2c0c2f07ad084a64f85debe9f6845b78f2) they work fine).

**Comment by rullzer** [comment]
> > Not sure what the acceptance tests complain about
> 
> The _app-comments_ can be ignored, they are caused by a regression in the sidebar. The others are legit; for some reason with this pull request the Apps app and the Users app show a blank page (and in [8f650fe^](https://github.com/nextcloud/server/commit/613f0f2c0c2f07ad084a64f85debe9f6845b78f2) they work fine).

which apps?

**Comment by rullzer** [comment]
aah the apps and user page
mmm indeed.... so I ahve a crude fix ut this seems to break js...

**Pull request merged**